### PR TITLE
Allowing creation of new context and new player

### PR DESCRIPTION
**Pull request by Sadzeih**
We ran into an issue where creating a new context and player caused a segmentation violation.

These modifications allow you to `player.Close` as you wish and make sure the `io.Copy` doesn't cause a segv.

### CODE REVIEW

File: player.go [hunk 1]
```diff
@@ -36,7 +36,6 @@ func newPlayer(context *Context) *Player {
 		r:       r,
 		w:       w,
 	}
-	runtime.SetFinalizer(p, (*Player).Close)
```
Comment (hajimehoshi) [inline player.go]: Why did you remove this? Without this, players can leak.

Comment (Sadzeih) [reply player.go]: Sure, but with the `SetFinalizer` when we need to close the Player manually before closing the Context or we get a `mux: already closed` error.

Comment (Sadzeih) [reply player.go]: And in my mind, it's the user's job to make sure the program doesn't leak, not the lib itself.

**Comment by Sadzeih** [comment]
I'm not sure, but it worked on my side. I closed the player, the context then recreated both and the music started playing again.

File: player.go (hunk 1 above)
Comment (hajimehoshi) [reply player.go]: > And in my mind, it's the user's job to make sure the program doesn't leak, not the lib itself.

Ensuring to call Close is a conventional way in Go (e.g., os.File).

Comment (hajimehoshi) [reply player.go]: > Sure, but we need to close the Player manually before closing the Context or we get a mux: already closed error.

If so, what we should do would be close all the players internally when closing the context.

**Comment by hajimehoshi** [comment]
Oh the document I wrote says

> Closing a context and opening a new one is allowed.

Then reopening the context should be fine :-)

File: player.go (hunk 1 above)
Comment (Sadzeih) [reply player.go]: > close all the players internally when closing the context

That's a better way IMO

Comment (hajimehoshi) [reply player.go]: OK, so please update the PR to do so :-)

**Comment by hajimehoshi** [comment]
In v2, we no longer close a context (and then we can never reopen a context).
